solax/inverter.py

```diff
@@ -287,6 +310,133 @@ def schema(cls):
         return cls.__schema
 
 
+def _energy(value, result):
+    value += result['Total Feed-in Energy Resets'] * 65535
+    value /= 100
+    return value
+
+
+def _consumption(value, result):
+    value += result['Total Consumption Resets'] * 65535
+    value /= 100
+    return value
+
+
+def _twoway_current(val, _):
+    return _to_signed(val, None) / 10
+
+
+def _div10(val, _):
+    return val / 10
+
+
+def _div100(val, _):
+    return val / 100
+
+
+def _to_signed(val, _):
+    if val > 32767:
+        val -= 65535
+    return val
+
+
+def _pv_power(_, result):
+    return result['PV1 Power'] + result['PV2 Power']
+
+
+def _load_power(_, result):
+    return result['AC Power'] - result['Exported Power']
```

Comment:
not sure if synthesizing values should be in this library. If things like the upstream home-assistant library require these values, the process of synthesizing the values should take place there.

Reply:
You're right (thanks for cleaning it up). I had added it because it's useful to have in HA (the docs even suggest it: https://www.home-assistant.io/integrations/solax/#optional-template-sensor), but I agree that the HA integration code is a better place for it.